src/providers/cryptoauthlib/hash.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::Provider;
+use parsec_interface::operations::psa_algorithm::Hash;
+use parsec_interface::operations::psa_hash_compute;
+use parsec_interface::requests::{ResponseStatus, Result};
+
+impl Provider {
+    pub(super) fn psa_hash_compute_internal(
+        &self,
+        op: psa_hash_compute::Operation,
+    ) -> Result<psa_hash_compute::Result> {
+        let mut hash = vec![0u8; op.alg.hash_length()];
+        let message = op.input.to_vec();
```

Comment:
Could you make sure that this buffer is zeroized if not sent to the `cryptoauthlib` library (i.e. in the `_` branch of the `match` below) ?

Reply:
Forgive me detailed question - you suggest to zeroize hash (the output) when algorithm is not supported by CAL/ATECC?

Reply:
No - to zeroize the `message` that you create on this line

Reply:
I admit I had an aesthetic problem with this zeroizing (or maybe the way I applied this?). I reduced the scope of the variable to a block where it is actually useful. Hope this addresses your issue as well.

Reply:
That works!

Just for future reference, what I had in mind was importing `zeroize::Zeroize` in the file and then calling [`message.zeroize()`](https://docs.rs/zeroize/1.2.0/zeroize/trait.Zeroize.html#tymethod.zeroize) in that `match` branch - is this what you tried?

Reply:
Yes, this was my first attempt. And that `message.zeroize()` looked for me as a workaround while narrowing down the scope fixed the issue by design.